### DISCUSSION TITLE
Zeros for "hull_junction_length" and "opening"

### DISCUSSION THREAD

**Discussion by Ssiimonn** [brainvisa/brainvisa.github.io#167] [Q&A]
Hi everyone,

I have two questions regarding the morphologist pipeline output.

(1) Having run the morphologist pipeline to obtain sulcal length, width, depth and surface, I noticed that for some scans, sulcal length and width of some sulci are zero. However, as these sulci do have non-zero values for sulcal depth and surface, it does not mean that the sulcus wasn't segmented for that scan at all. This seems to suggest to me that the software has a hard time determining what is the width and what is the length of these sulci. I've looked into which sulci have zeros for length and width most often, and those sulci are generally more "circular" shaped, which supports my reasoning.

Then again, I'm relatively new to brainVISA so I might as well be way off here. Could anyone confirm my thoughts, or if my theory is incorrect, could anyone tell me what is the reason for these zeros for the length and width of some sulci?

(2) Can I safely assume that, when sulcal length, width, depth and surface are **all missing** of a certain sulcus for a certain scan, this is because the sulcus was **not** segmented at all for this brain, and thus simply not present?

Kind regards,

Simon

**Comment by denisri** [comment]
Hi,
(1) The length of a sulcus is calculated by measuring the length of the junction line between the sulcus and the external brain hull. Thus, this works when the sulcus actually touches the external hull. But a few burried sulci, like the insula, do not touch it, thus we don't have a length measurement for them. If you observe this on external sulci, could you please provide me with an example ?

(2) Yes that's it: when a sulcus is not found (missing or not identified as this sulcus), then most morphometric values are set to 0.

Denis

**Comment by Ssiimonn** [comment]
Hi Denis,

Thank you for your response.

Before I provide you with (an) example(s), I need to know exactly the difference between when the value becomes NA, and when the value is set to 0.

You say morphometric values are set to 0 when a sulcus is not present, but what exactly is the difference between an NA and a zero in the dataset? I thought the values were set to NA (and not 0) when a sulcus is not present.

The reason I ask is because I'd like to know the difference between:

- when a sulcus was simply not "present" in an individual's brain (i.e., this is valuable information that I shouldn't just disregard as measurement error, because it could be group-related) -- I think this is when it's NA/missing?
- and when a sulcus **was** detected (and therefore "exists" in this person's brain) but a certain measurement just couldn't be performed due to, for example, the reason you describe above in point (1) -- I think this is when it's 0?

**Comment by denisri** [comment]
I suppose you have obtained the features using the brainvisa process "sulci graph morphometry by subject", is that right ?
This process sets values to NaN in some situations, which are, if I'm reading the code correctly, when a sulcus is not found in the subject data (thus, missing or at least not identified), which contradicts what I have said earlier (because I was looking in a lower-level code, which actually sets 0, and the older process, "morphometry" does not the same actually). So yes you're right. Now if there are cases you still get zeros for all features, I'm a bit surprised right now but I haven't looked at the code thoroughly yet.

**Reply by Ssiimonn** [reply]
Awesome, that clears things up. I actually don't think there are cases in my dataset where there's a zero for all measurements; only cases with NaNs for all measurements. I can confirm later.

Thanks a bunch, Denis!

**Reply by Ssiimonn** [reply]
Right, I promised I'd confirm later, so I hereby confirm that there's **no** cases where I get zeros for all four features.

As a last question (hopefully), having replaced the empty cells with "unidentified" where applicable now, I am left with some empty cells here and there (so **not** zero, but really missing) in my dataset for mean depth specifically. Have you heard of that happening before? Could you tell me how depth is computed as well, perhaps?

**Comment by denisri** [comment]
Hi,
Mean depth is computed as the average of the geodesic depth of the bottom line points of the sulcus (the deepest voxels). It could be missing in individual sulcal graph nodes if there were no bottom point at all (even if I doubt this can happen), and in the process "sulci graph morphometry by subject" it should be always present, potentially 0 if no bottom point is found (unlikely I think). The output of this process is a table with fixed columns, so I don't understand how a cell could be missing...
Denis

**Reply by Ssiimonn** [reply]
Thank you for your input again. In case it's of any help, I've marked the sulci in which it happens for one or two scans.

![image](https://github.com/brainvisa/brainvisa.github.io/assets/65156426/bdcc5d6c-4adb-4328-8d83-80f9c4ebae55)
